documentation/CAMARA-OIDC-profile.md

```diff
@@ -0,0 +1,1022 @@
+
+    * **_invalid_grant_**
+
+      The provided authorization grant authorization code is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client.
+
+    * **_unauthorized_client_**
+
+      The authenticated client is not authorized to use this authorization grant type.
+
+    * **_unsupported_grant_type_**
+
+      The authorization grant type is not supported by the authorization server.
+
+    * **_invalid_scope_**
+
+      The requested scope is invalid, unknown, malformed, or exceeds the scope granted by the resource owner.
+
+
+* **error_description**
+
+  OPTIONAL. Human-readable ASCII text providing additional information, used to assist the client developer in understanding the error that occurred. Values for the `error_description` parameter MUST NOT include characters outside the set %x20-21 / %x23-5B / %x5D-7E.
+
+
+The parameters are included in the entity-body of the HTTP response using the `application/json` media type.
+
+`error_description` parameter is not a valid field to take decisions on the developer side, as its value may vary depending on the server implementation, language, format or context. Therefore, the developer should base their logic only on the `error` parameter.
+
+The following is a non-normative example Token Error Response:
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+```
+```json
+{
+  "error": "unauthorized_client"
+}
+```
+
+## The Scope Parameter
+
+Oauth scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services.
+
+The Authorization Request MUST contain the `scope` parameter, which is a space delimited, case-sensitive list of ASCII strings (scope values). Multiple scope values can be requested simultaneously, subject to the SP being registered to use those “scopes”.
+
+For OpenID requests, the scope values MUST include `openid`, to indicate that the request is an OpenID Connect request, followed by other values depending on the specific CAMARA services being requested.
+
+A purpose must be declared within the requested scope for the following grant types:
+
+* _authorization_code_
+
+* _urn:openid:params:grant-type:ciba_
+
+No purpose is required for the `client_credentials` grant type, as this grant type can only be used when no personal user data is processed. The requested scope must be set directly to the value defined for the relevant endpoint within the OAS ("YAML") specification. "Wild card" scopes (i.e. specifying only the API name) are not valid for this grant type.
+
```

Comment:
Even for client_credential, a purpose MUST be sent. An API client must sign up for using a certain API for a certain purpose. The purpose might be charging relevant. If a client has signed up for the same API, but different purposes, it will not be possible to detect the correct monetization / charging rule, unless the purpose is sent.

Reply:
The `purpose` is not intended to support different charging models. If an API provider wishes to charge a given customer different amounts for the same API depending on why the customer is calling that API, then they must do that by allocating different client ids to that customer.

CAMARA should not attempt to standardise "use cases". Whilst an intended use case is important for API design, restricting use cases to a defined set is fatal for any ecosystem claiming to be "open".